### ISSUE TITLE
Seppe Vansteelant: PersonalData and DataRetention should be linked to PersonalDataHandling and Processing.

### ISSUE THREAD

**Issue by michaelgeamanu** [Informatievlaanderen/OSLOthema-consent#38]
Het feit dat de entiteiten PersonalData en DataRetentiondata niet gerelateerd zijn aan de verwerkingsactiviteit (dus nu PersonalDataHandling en Processing) lijkt met niet te kloppen. Er zijn namelijk andere voorwaarden die de bewaartermijn kunnen bepalen dan de toestemming, bovendien hangt die retentieperiode echt wel samen met de verwerkingsactiviteit. Ook de ‘PersonalData’ staat daar niet helemaal juist volgens mij. Je geeft toestemming voor een verwerkingsactiviteit. Het doeleinde bepaalt welke persoonsgegevens je daarvoor nodig hebt, dus wat vreemd dat dit enkel aan de entiteit consent hangt.

**Comment by GeertThijs** [comment]
You are partly right. Core of the matter is that we mixed two approaches on Consent in one application profile, thinking to have best of both worlds but actually ending up with a model that leads to confusion. These approaches are:

- [GConsent](https://openscience.adaptcentre.ie/ontologies/GConsent/docs/ontology): revolves around the class Consent, which is actually an instance of a DPV:LegalBasis of type Consent.
- [DPV](https://w3c.github.io/dpv/dpv/): central class here is PersonalDataHandling, which focusses on the event of the data handling rather than at the Consent (or other LegalBases) for the data handling which is only one aspect of it.

As the focus of this standard was on the Consents we chose the first approach but complemented it with a link with the class PersonalDataHandling to take advantage of the fact that it groups certain aspects like Processing, Purpose etc. In combining both models there were al lot of redundant relations that exested as well for Consent as for PersonalDataHandling (eg hasDataSubject, hasDataController, hasProcessing etc). To simplify things we removed from PersonalDataHandling certain relations that were already provided in GConsent en kept other relations that we thought were better covered by this class.

The result is indeed somewhat confusing: eg hasDataSubject is a feature of Consent as well as from PersonalDataHandling but we only kept Consent.hasDataSubject. The relations hasProcessing and forProcessing occur for PersonalDataHandling as well as for Consent respectively but we chose to retain only PersonalDataHandling.hasProcessing.

From the viewpoint of Consent this poses no implementation issues however, if one relation or the other is used does not matter as there is a one-on-one relation between Consent and PersonalDataHandling. So if Processing is associated with Consent or with PersonalDataHandling: the result is the same. Your remark that PersonalData and DataRetentions should be associated with Processing rather than with Consent/PersonaldataHandling is a seperate issue and is in fact an issue with GConsent and DPV that we we feel hesitant to correct ourselves in OSLO Consent.

If later on other kinds of LegalBases should be processed the viewpoint of PersonalDataHandling should be preferrred over this of Consent and the current way we modelled OSLO Consent cannot be used as is. But that is another issue.

**Comment by GeertThijs** [comment]
You are correct. All these issues are solved if we would remap the current model to DPV (see more on this in issue #46). Purpose is then connected to PersonalDataHandling, while data retention (called StorageCondition in DPV) is in Processing. And indeed, PersonalData is connected to PersonalDataHandling. In DPV, PersonalDataHandling is the class that groups all aspects of it, including LegalBasis (like Consent).

**Comment by SeppeVansteelant** [comment]
ok, looks better right now apart from the term PersonalDataHandling, see issue #37
